### DISCUSSION TITLE
Features on "backpack/calendar-operation"

### DISCUSSION THREAD

**Discussion by ShariqAyaz** [Laravel-Backpack/community-forum#1036] [Ideas]
Adding followings may helps:
1. Off Days or Day so cannot book anything + Business Timing
<img width="841" alt="image" src="https://github.com/Laravel-Backpack/community-forum/assets/22431541/6bde9302-55e0-4e13-91e1-d0b23b5162be">

2. Enumeration Bind with Predefined colours.
<img width="277" alt="image" src="https://github.com/Laravel-Backpack/community-forum/assets/22431541/a539adb4-3399-4f57-ac6d-960d98cfd616">
4. Or somehow accommodate 'type of slot' and represents colors wight now each time need to change color.
5. Concatenation wihin "title" maybe demanding for most users as I can see I want to put two database field as title.
```
    protected function getCalendarFieldsMap()
    {
        return [
            'title' => $a.$b.$c, // example
            'start' => 'start',
            'end' => 'end',
            'background_color' => 'background_color',
            'text_color' => 'text_color',
            'all_day' => 'all_day',
        ];
    }
```

   ```
// HERE
 public function setupCalendarOperation()
    {
        CRUD::setOperationSetting('initial-view', 'dayGridMonth');
        CRUD::setOperationSetting('views', ['dayGridMonth', 'timeGridWeek', 'timeGridDay']);
        CRUD::setOperationSetting('editable', true);
        CRUD::setOperationSetting('background_color', fn ($event) => $event->active ? 'green' : 'red');
        CRUD::setOperationSetting('text_color', fn ($event) => $event->active ? 'white' : 'black');
        CRUD::setOperationSetting('with-javascript-widget', true);
    }
```

**Comment by jcastroa87** [comment]
Hello @ShariqAyaz 

You can check [FullCalendar docs](https://fullcalendar.io/docs/) like [weekends](https://fullcalendar.io/docs/weekends) maybe and add as an option in calendar operation

```
public function setupCalendarOperation()
{
    CRUD::setOperationSetting('javascript-configuration', [
        // ... FullCalendar options
        // https://fullcalendar.io/docs
        'dayMaxEvents' => false, // by default backpack uses 5
    ]);
}
```

Read carefully [our docs](https://backpackforlaravel.com/products/calendar-operation).

Cheers.

**Comment by ShariqAyaz** [comment]
One Issue solved:
```
        $start = request()->has('start') ? Carbon::parse(request('start'))->setTimezone(config('app.timezone')) : null;
        $end = request()->has('end') ? Carbon::parse(request('end'))->setTimezone(config('app.timezone')) : null;
```

**Comment by jcastroa87** [comment]
Hello @ShariqAyaz as i understand you solve this issue. So i will close it, but please feel free to re-open or create a new one if needed.

Cheers.
